api/src/DataProvider/InvitationDataProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Core\DataProvider\SerializerAwareDataProviderTrait;
+use App\DTO\Invitation;
+use App\Repository\CampCollaborationRepository;
+use App\Repository\UserRepository;
+use Symfony\Component\Security\Core\Security;
+
+class InvitationDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface {
+    use SerializerAwareDataProviderTrait;
+    private CampCollaborationRepository $campCollaborationRepository;
+    private Security $security;
+    private UserRepository $userRepository;
+
+    /**
+     * InvitationDataProvider constructor.
+     */
+    public function __construct(Security $security, UserRepository $userRepository, CampCollaborationRepository $campCollaborationRepository) {
+        $this->security = $security;
+        $this->userRepository = $userRepository;
+        $this->campCollaborationRepository = $campCollaborationRepository;
+    }
+
+    public function getItem(string $resourceClass, $id, string $operationName = null, array $context = []): ?Invitation {
+        if (null === $id) {
+            return null;
+        }
+        $campCollaboration = $this->campCollaborationRepository->findByInviteKey($id);
+        if (null === $campCollaboration) {
+            return null;
+        }
+        $username = null;
+        $userAlreadyInCamp = null;
+
+        $camp = $campCollaboration->camp;
+        if (null !== $this->security->getUser()) {
+            $username = $this->security->getUser()->getUserIdentifier();
+            $user = $this->userRepository->findOneBy(['username' => $username]);
```

Comment:
Gibts einen konkreten Grund warum du hier zuerst den Usernamen (`getUserIdentifier`) aus dem User rausholst, und dann wieder den User mithilfe des Usernamen in der Datenbank suchst?

Reply:
Ja, wenn ich nach dem Resultat von getUser (Was UserInterface ist) suche, findet Doctrine die existing CampCollaboration nicht.